flight/UAVTalk/uavtalk.c

```diff
@@ -368,14 +368,14 @@ UAVTalkRxState UAVTalkProcessInputStreamQuiet(UAVTalkConnection connectionHandle
 		iproc->obj = UAVObjGetByID(iproc->objId);
 
 		// Determine data length
+		iproc->timestampLength = (iproc->type & UAVTALK_TIMESTAMPED) ? 2 : 0;
 		if (iproc->type == UAVTALK_TYPE_OBJ_REQ || iproc->type == UAVTALK_TYPE_ACK || iproc->type == UAVTALK_TYPE_NACK) {
```

Comment:
Note that, for instance, this is already broken for timestamps too, because it doesn't mask off UAVTALK_TIMESTAMPED or list the other possibilities.